### PR TITLE
test_scalers: Add more assertions

### PR DESCRIPTION
**Pull request by auscompgeek**
- Test all the boundary conditions
- Check the result is always between 0 and the given value for deadzone and exponential
- Test that deadzone and exponential are odd functions

### CODE REVIEW

File: tests/test_scalers.py [hunk 1]
```diff
@@ -10,21 +10,30 @@
...
+    elif value > 0:
+        assert 0 < result < value
+    else:
+        assert value < result < 0
 
 
-@given(value=floats(-1, 1), base=floats(1, exclude_min=True, allow_infinity=False))
+@given(
+    value=floats(-1, 1),
+    base=floats(1, exclude_min=True, allow_infinity=False, width=16),
```
Comment (auscompgeek) [inline tests/test_scalers.py:27]: Restricting the required precision here to avoid compounding floating point error for floats close to 1. Given that we will usually use "round" numbers for the exponential base, I think this is reasonable.

**Review by james-ward — APPROVED** [review]
